application/src/main/java/org/togetherjava/tjbot/commands/help/HelpThreadOverviewUpdater.java

```diff
@@ -0,0 +1,149 @@
+package org.togetherjava.tjbot.commands.help;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.*;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.requests.RestAction;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.MessageReceiverAdapter;
+import org.togetherjava.tjbot.commands.Routine;
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.config.HelpSystemConfig;
+
+import java.awt.Color;
+import java.util.List;
+import java.util.Optional;
+import java.util.StringJoiner;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+// FIXME Javadoc
+public final class HelpThreadOverviewUpdater extends MessageReceiverAdapter implements Routine {
+    private static final Logger logger = LoggerFactory.getLogger(HelpThreadOverviewUpdater.class);
+
+    private static final Color AMBIENT_COLOR = Color.decode("#CCCC00");
+
+    private final HelpSystemConfig config;
+    private final Predicate<String> isOverviewChannelName;
+    private final Predicate<String> isStagingChannelName;
+
+    /**
+     * Creates a new instance.
+     *
+     * @param config the config to use
+     */
+    public HelpThreadOverviewUpdater(@NotNull Config config) {
+        super(Pattern.compile(config.getHelpSystem().getStagingChannelPattern()));
+
+        this.config = config.getHelpSystem();
+        isOverviewChannelName = Pattern.compile(config.getHelpSystem().getOverviewChannelPattern())
+            .asMatchPredicate();
+        isStagingChannelName = Pattern.compile(config.getHelpSystem().getStagingChannelPattern())
+            .asMatchPredicate();
+    }
+
+    @Override
+    public @NotNull Schedule createSchedule() {
+        return new Schedule(ScheduleMode.FIXED_RATE, 1, 1, TimeUnit.MINUTES);
+    }
+
+    @Override
+    public void runRoutine(@NotNull JDA jda) {
+        jda.getGuilds().forEach(this::updateOverview);
```

Comment:
```suggestion
        jda.getGuildCache().forEach(this::updateOverview);
```

To refresh you:

getGuilds = new ArrayList<>(getGuildCache);

getGuilds does NOT retrieve from Discord, it just creates a list from the cache
getGuildCache doesn't make a list, so you can't make any changes

Reply:
beat me to it, I will always overlook this 😆